binrw/src/lib.rs

```diff
@@ -1,148 +1,4 @@
-//! [mapping values](attribute#map).
-//!
-//! Unlike “zero-copy” libraries, the in-memory representation of binrw structs
-//! doesn’t need to match the raw data. This can allow for better memory
-//! performance, especially on architectures where unaligned memory access is
-//! slow. Also, because data is never [transmuted](core::mem::transmute), there
-//! is no risk of undefined behaviour.
-//!
-//! # Input
-//!
-//! `BinRead` reads data from any object that implements [`io::Read`] +
-//! [`io::Seek`]. This means that data can come from memory, network, disk, or
-//! any other streaming source. It also means that low-level data operations
-//! like buffering and compression are efficient and easy to implement.
-//!
-//! `BinRead` also includes an [extension trait](BinReaderExt) for reading types
-//! directly from input objects:
-//!
-//! ```rust
-//! use binrw::{BinReaderExt, io::Cursor};
-//!
-//! let mut reader = Cursor::new(b"\x00\x0A");
-//! let val: u16 = reader.read_be().unwrap();
-//! assert_eq!(val, 10);
-//! ```
-//!
-//! # Directives
-//!
-//! Handling things like magic numbers, byte ordering, and padding & alignment
-//! is typical when working with binary data, so binrw includes a variety of
-//! [built-in directives](attribute) for these common cases that can be applied
-//! using the `#[br]` attribute:
-//!
-//! ```
-//! # use binrw::{prelude::*, io::Cursor, NullString};
-//! #
-//! #[derive(BinRead)]
-//! #[br(magic = b"DOG", assert(name.len() != 0))]
-//! struct Dog {
-//!     bone_pile_count: u8,
-//!
-//!     #[br(big, count = bone_pile_count)]
-//!     bone_piles: Vec<u16>,
-//!
-//!     #[br(align_before = 0xA)]
-//!     name: NullString
-//! }
-//!
-//! let mut data = Cursor::new(b"DOG\x02\x00\x01\x00\x12\0\0Rudy\0");
-//! let dog = Dog::read(&mut data).unwrap();
-//! assert_eq!(dog.bone_piles, &[0x1, 0x12]);
-//! assert_eq!(dog.name.to_string(), "Rudy")
-//! ```
-//!
-//! Directives can also reference earlier fields by name. For tuple types,
-//! earlier fields are addressable by `self_N`, where `N` is the index of the
-//! field.
-//!
-//! See the [attribute module](attribute) for the full list of available
-//! directives.
-//!
-//! # Built-in implementations
-//!
-//! Implementations for all primitive data types, arrays, tuples, and standard
-//! Rust types like [`Vec`] are included, along with parsers for other
-//! frequently used binary data patterns like
-//! [null-terminated strings](NullString) and
-//! [indirect addressing using offsets](FilePtr). Convenient access into
-//! bitfields is possible using crates like
-//! [modular-bitfield](attribute#using-map-on-a-struct-to-create-a-bit-field).
-//!
-//! See the [`BinRead` trait](BinRead#foreign-impls) for the full list of
-//! built-in implementations.
-//!
-//! # no_std support
-//!
-//! binrw supports no_std and includes a compatible subset of [`io`]
-//! functionality. The [`alloc`] crate is required.
-
+#![doc = include_str!("../doc/index.md")]
```

Comment:
I was going to ask if this was compatible with our MSRV policy but realized we don't have one.

I'm definitely a fan of this but fair warning might revert down the road if it is the highest MSRV blocker by more than ~1 release.